SimG4CMS/Calo/src/HFDarkening.cc

```diff
@@ -5,145 +5,64 @@
 
-float HFDarkening::dose(int layer, float Radius) {
-  //	float dose_acquired = 0.;
-  int radius = (int) floor((Radius-12.5)/10.0+0.5);
-  if (layer == 15) {
+double HFDarkening::dose(unsigned int layer, double Radius) {
+  // Radii are 13-17, 17-20, 20-24, 24-29, 29-34, 34-41, 41-48, 48-58, 58-69, 69-82, 82-98, 98-116, 116-130
+  // These radii are specific to the geometry of the dose map, which closely matches HF Tower Geometry,
+  // but not exactly.
+  if (layer > (_numberOfZLayers-1))
+  {
     return 0.;
   }
-  return dose_layer_depth[layer][radius];
-
-  /*
-  for (int i = 0; i<= layer; ++i) {
-    dose_acquired+=dose_layer_depth[i][radius];
-  }
-	
-  if((layer<-1)||(layer>17)) return (0.);
-
-  if((layer==-1)||(layer==0)) {
-    if(ind<0) return (dose_layer_depth[0][0]); if(ind>=9) return (dose_layer_depth[0][10]);
-    if((ind>=0)&&(ind<=9)) {float delta = (dose_layer_depth[0][ind+1]-dose_layer_depth[0][ind])/10; return (dose_layer_depth[0][ind]+delta*(Radius-radius[ind]));}
-  }
-  if((layer>=1)&&(layer<=3)) {
-    if(ind<0) return (dose_layer_depth[5][0]); if(ind>=9) return (dose_layer_depth[5][10]);
-    if((ind>=0)&&(ind<=9)) {float delta = (dose_layer_depth[5][ind+1]-dose_layer_depth[5][ind])/10; return (dose_layer_depth[5][ind]+delta*(Radius-radius[ind]));}
-  }
-  if((layer==4)||(layer==5)) {
-    if(ind<0) return (dose_layer_depth[10][0]); if(ind>=9) return (dose_layer_depth[10][10]);
-    if((ind>=0)&&(ind<=9)) {float delta = (dose_layer_depth[10][ind+1]-dose_layer_depth[10][ind])/10; return (dose_layer_depth[10][ind]+delta*(Radius-radius[ind]));}
-  }
-  if((layer>=6)&&(layer<=8)) {
-    if(ind<0) return (dose_l6_l8[0]); if(ind>=23) return (dose_l6_l8[23]);
-    if((ind>=0)&&(ind<=23)) {float delta = (dose_l6_l8[ind+1]-dose_l6_l8[ind])/10; return (dose_l6_l8[ind]+delta*(Radius-radius[ind]));}
-  }
-  if((layer==9)||(layer==10)) {
-    if(ind<0) return (dose_l9_l10[0]); if(ind>=23) return (dose_l9_l10[23]);
-    if((ind>=0)&&(ind<=23)) {float delta = (dose_l9_l10[ind+1]-dose_l9_l10[ind])/10; return (dose_l9_l10[ind]+delta*(Radius-radius[ind]));}
-  }
-  if((layer>=11)&&(layer<=13)) {
-    if(ind<0) return (dose_l11_l13[0]); if(ind>=23) return (dose_l11_l13[23]);
-    if((ind>=0)&&(ind<=23)) {float delta = (dose_l11_l13[ind+1]-dose_l11_l13[ind])/10; return (dose_l11_l13[ind]+delta*(Radius-radius[ind]));}
-  }
-  if((layer==14)||(layer==15)) {
-    if(ind<0) return (dose_l14_l15[0]); if(ind>=23) return (dose_l14_l15[23]);
-    if((ind>=0)&&(ind<=23)) {float delta = (dose_l14_l15[ind+1]-dose_l14_l15[ind])/10; return (dose_l14_l15[ind]+delta*(Radius-radius[ind]));}
-  }
-  if((layer==16)||(layer==17)) {
-    if(ind<0) return (dose_l16_l17[0]); if(ind>=23) return (dose_l16_l17[23]);
-    if((ind>=0)&&(ind<=23)) {float delta = (dose_l16_l17[ind+1]-dose_l16_l17[ind])/10; return (dose_l16_l17[ind]+delta*(Radius-radius[ind]));}
-  }
-  return 0.;
-  */
+  
+  int radius = 0;
+  if (Radius > 13.0 && Radius <= 17.0) radius = 0;
```

Comment:
@jwwetzel - please rework this to use else ifs - which remove the need to check a range of radii rather than just the maximum (and will dramatically reduce the number of comparisons made. 

Eg

if ( Radius <= 17.f) radius= 0;
else if ( Radius <= 20.0) radius=1;
else if....